### PR TITLE
Fix reference tests

### PR DESCRIPTION
**Pull request by shans**
Note that this does cause some other tests to fail :(

### CODE REVIEW

File: src/runtime/particle-execution-host.ts [hunk 1]
```diff
@@ -283,18 +285,31 @@ class PECOuterPortImpl extends PECOuterPort {
...
+        throw new Error(`Don't know how to invent new storage keys for new storage stack when we only have type information`);
+      }
+      const key = StorageKeyParser.parse(storageKey);
+      // TODO(shanestephens): We could probably register the active store here, but at the moment onRegister and onProxyMessage both
+      // expect to be able to do activation
+      const storeBase = new Store({id: storageKey, exists: Exists.MayExist, storageKey: key, type});
+      this.GetBackingStoreCallback(storeBase, callback, type, type.toString(), storageKey, storageKey);
+    } else {
+      if (!storageKey) {
+        // XXX
```
Comment (mykmartin) [inline src/runtime/particle-execution-host.ts]: ?

Comment (shans) [reply src/runtime/particle-execution-host.ts]: This was in the old code too.. :( It's gonna get deleted once we switch across to the new stack.

File: src/runtime/reference.ts [hunk 2]
```diff
@@ -87,13 +94,24 @@ export abstract class ClientReference extends Reference {
...
+        entityStorageKey: Flags.useNewStorageStack ? Entity.storageKey(entity) : null
+      },
+      new ReferenceType(Entity.entityClass(entity).type),
+      context
+    );
 
     this.entity = entity;
-    this.stored = this.storeReference(entity);
+    if (Flags.useNewStorageStack) {
+      this.stored = Promise.resolve();
```
[the remark this answers is not in the document]
Comment (shans) [reply src/runtime/reference.ts]: No, storeReference is used in the old stack to write the entity back to a backing store. With the new stack it's required that the entity already be stored somewhere and we don't make a copy.

**Comment by shans** [comment]
Manual copybara run at cl/290537578

**Pull request merged**